core/src/main/java/org/infinispan/security/AuthorizationManager.java

```diff
@@ -27,4 +28,11 @@
     * has the requested permission and role. A {@link SecurityException} is thrown otherwise.
     */
    void checkPermission(AuthorizationPermission permission, String role);
+
+   /**
+    * Verifies that the {@link Subject} associated with the current {@link AccessControlContext}
+    * has the requested permission and at least one of roles.
+    * A {@link SecurityException} is thrown otherwise.
+    */
+   void checkPermission(AuthorizationPermission perm, Collection<String> roles);
```

Comment:
Maybe name the method to clearly indicate that it is just going to need at least one matching role, because otherwise it looks like it implies it needs all roles